src/routes/+layout.ts

```diff
@@ -1,15 +1,53 @@
-import { userInfo, validateCredsAndFillUserInfo } from "$lib/utils/auth";
+import { browser } from "$app/environment";
+import {
+  disconnect,
+  userInfo,
+  validateCredsAndFillUserInfo,
+} from "$lib/utils/auth";
 import { redirect } from "@sveltejs/kit";
+import dayjs from "dayjs";
 import { get } from "svelte/store";
 import type { LayoutLoad } from "./$types";
 
 export const prerender = false;
-export const load: LayoutLoad = async ({ url }) => {
+
+function tokenWillExpireSoon(tokenExpirationString: string): boolean {
+  const tokenExpirationDate = dayjs(tokenExpirationString);
+  if (!tokenExpirationDate.isValid()) {
+    // Impossible de parser la date d'expiration -- on considère que
+    // le token est invalide
+    return true;
+  }
+  const minimalValidityDate = dayjs().add(1, "days");
+  return tokenExpirationDate.isBefore(minimalValidityDate);
+}
+
+export const load: LayoutLoad = async ({ url, depends }) => {
+  if (!browser) {
+    // On ne fait pas d'authentification coté serveur
+    // => on peut court-circuiter les vérifications ici
+    return {};
+  }
+
+  // On veut vérifier le token à chaque changement de page
+  depends(url.pathname);
```

Comment:
Tu es sur du comportement de `depends` ?

Si j'en crois la doc https://kit.svelte.dev/docs/types#public-types-loadevent, `depends` doit être utilisé conjointement avec `invalidate` pour fonctionner - et je ne vois pas de `invalidate` dans le code

```
It declared a dependency on a specific URL via [fetch](https://kit.svelte.dev/docs/load#making-fetch-requests) or [depends](https://kit.svelte.dev/docs/types#public-types-loadevent), and that URL was marked invalid with [invalidate(url)](https://kit.svelte.dev/docs/modules#$app-navigation-invalidate)
```

Toutefois, la doc évoque aussi que le `load` rerun si `It references a property of url (such as url.pathname or url.search) whose value has changed`

Du coup, il est possible que le `depends` soit superflu (et aurait pu être remplacé par n'importe quoi) mais qu'on obtienne le résultat escompté car le `depends` a comme paramètre `url.pathname` 🙂

Et je veux juste vérifier si c'est fait exprès ou non :)

Reply:
Bien vu! Un simple console.log(url.pathname) faisait la même chose.
Du coup, j'ai revu le code pour rendre ça plus explicite, et en bonus pour ne recharger la fonction que si l'utilisateur est connecté. Dis-moi si ça te parait clair!